ports/esp32s2/common-hal/alarm/time/TimeAlarm.c

```diff
@@ -37,3 +41,63 @@ void common_hal_alarm_time_time_alarm_construct(alarm_time_time_alarm_obj_t *sel
 mp_float_t common_hal_alarm_time_time_alarm_get_monotonic_time(alarm_time_time_alarm_obj_t *self) {
     return self->monotonic_time;
 }
+
+mp_obj_t alarm_time_timealarm_get_wakeup_alarm(size_t n_alarms, const mp_obj_t *alarms) {
+    // First, check to see if we match
+    for (size_t i = 0; i < n_alarms; i++) {
+        if (MP_OBJ_IS_TYPE(alarms[i], &alarm_time_time_alarm_type)) {
+            return alarms[i];
+        }
+    }
+    alarm_time_time_alarm_obj_t *timer = m_new_obj(alarm_time_time_alarm_obj_t);
+    timer->base.type = &alarm_time_time_alarm_type;
+    return timer;
```

Comment:
Did you want to set the `time_monotonic` value here so it can be read by the user looking at the alarm?
Or maybe we don't know what the value is.

Reply:
Added a TODO. This PR is huge and I feel like I'm close to merging it. :-)